Modules/Devices/src/Devices.Application/Identities/Commands/UpdateIdentity/Handler.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Linq;
+using Backbone.Modules.Devices.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Devices.Application.IntegrationEvents.Outgoing;
+using Backbone.Modules.Devices.Domain.Aggregates.Tier;
+using Backbone.Modules.Devices.Domain.Entities;
+using Enmeshed.BuildingBlocks.Application.Abstractions.Exceptions;
+using Enmeshed.BuildingBlocks.Application.Abstractions.Infrastructure.EventBus;
+using MediatR;
+using ApplicationException = Enmeshed.BuildingBlocks.Application.Abstractions.Exceptions.ApplicationException;
+
+namespace Backbone.Modules.Devices.Application.Identities.Commands.UpdateIdentity;
+public class Handler : IRequestHandler<UpdateIdentityCommand, Identity>
+{
+    private readonly IIdentitiesRepository _identitiesRepository;
+    private readonly ITiersRepository _tiersRepository;
+    private readonly IEventBus _eventBus;
+
+    public Handler(IIdentitiesRepository identitiesRepository, ITiersRepository tiersRepository, IEventBus eventBus)
+    {
+        _identitiesRepository = identitiesRepository;
+        _tiersRepository = tiersRepository;
+        _eventBus = eventBus;
+    }
+
+    public async Task<Identity> Handle(UpdateIdentityCommand request, CancellationToken cancellationToken)
+    {
+        var newTierId = TierId.Create(request.TierId);
```

Comment:
This is a result, not a tier id...
```suggestion
        var newTierIdResult = TierId.Create(request.TierId);
```